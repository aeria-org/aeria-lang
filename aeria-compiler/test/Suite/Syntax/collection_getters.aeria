collection CollectionGettersExample {
  getters {
    foo @js (doc) => {
      return "foo"
    }
    @end
    bar @js (doc) => {
      if (){}
      return "bar"
    }
    @end
  }
}
