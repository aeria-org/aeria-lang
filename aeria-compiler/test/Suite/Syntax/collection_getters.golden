{
  "kind": "Program",
  "collections": [
    {
      "name": {
        "value": "collectiongettersexample",
        "kind": "CollectionName"
      },
      "kind": "Collection",
      "getters": [
        {
          "name": {
            "value": "foo",
            "kind": "PropertyName"
          },
          "macro": {
            "macro": "{\n      return \"foo\"\n    }\n    ",
            "kind": "Macro"
          },
          "kind": "Getter"
        },
        {
          "name": {
            "value": "bar",
            "kind": "PropertyName"
          },
          "macro": {
            "macro": "{\n      if (){}\n      return \"bar\"\n    }\n    ",
            "kind": "Macro"
          },
          "kind": "Getter"
        }
      ]
    }
  ]
}
